### PR TITLE
[SCB-2663]Support to update microservice properties to the registration center.

### PR DESCRIPTION
**Pull request by Vientu**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install -Pit` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: clients/service-center-client/src/main/java/org/apache/servicecomb/service/center/client/model/UpdatePropertiesRequest.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.servicecomb.service.center.client.model;
+
+import java.util.Map;
+
+/**
+ * 更新微服务和实例properties的request
```
Comment (Shoothzj) [inline clients/service-center-client/src/main/java/org/apache/servicecomb/service/center/client/model/UpdatePropertiesRequest.java:23]: we should use english here

File: clients/service-center-client/src/test/java/org/apache/servicecomb/service/center/client/ServiceCenterClientTest.java [hunk 2]
```diff
@@ -489,4 +490,23 @@ public void TestUpdateServiceSchema() throws IOException {
 
     Assertions.assertTrue(result);
   }
+
+  @Test
+  public void TestUpdateMicroserviceProperties() throws IOException {
+
```
Comment (Shoothzj) [inline clients/service-center-client/src/test/java/org/apache/servicecomb/service/center/client/ServiceCenterClientTest.java:496]: no meaning blank line

File: demo/demo-cse-v2/provider/src/main/resources/application.yml [hunk 3]
```diff
@@ -28,11 +28,11 @@ servicecomb:
     name: provider
     version: 0.0.1
     registry:
-      address: ${CSE_V2_SC}
+      address: http://127.0.0.1:30100
```
Comment (Shoothzj) [inline demo/demo-cse-v2/provider/src/main/resources/application.yml:31]: is that change intentionally?

File: clients/service-center-client/src/main/java/org/apache/servicecomb/service/center/client/model/UpdatePropertiesRequest.java (hunk 1 above)
Comment (Vientu) [reply clients/service-center-client/src/main/java/org/apache/servicecomb/service/center/client/model/UpdatePropertiesRequest.java]: ok

File: clients/service-center-client/src/test/java/org/apache/servicecomb/service/center/client/ServiceCenterClientTest.java (hunk 2 above)
Comment (Vientu) [reply clients/service-center-client/src/test/java/org/apache/servicecomb/service/center/client/ServiceCenterClientTest.java]: I've deleted it.

File: demo/demo-cse-v2/provider/src/main/resources/application.yml (hunk 3 above)
Comment (Vientu) [reply demo/demo-cse-v2/provider/src/main/resources/application.yml]: No, i have rolled back the changes

File: clients/service-center-client/src/test/java/org/apache/servicecomb/service/center/client/ServiceCenterClientTest.java [hunk 4]
```diff
@@ -489,4 +490,22 @@ public void TestUpdateServiceSchema() throws IOException {
 
     Assertions.assertTrue(result);
   }
+
+  @Test
+  public void TestUpdateMicroserviceProperties() throws IOException {
```
Comment (liubao68) [inline clients/service-center-client/src/test/java/org/apache/servicecomb/service/center/client/ServiceCenterClientTest.java:495]: Use `testXXXX` for method name

File: service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/task/MicroserviceRegisterTask.java [hunk 5]
```diff
@@ -86,6 +86,24 @@ protected boolean doRegister() {
     if (!StringUtils.isEmpty(serviceId)) {
       // This microservice has been registered, so we just use the serviceId gotten from service center
       microservice.setServiceId(serviceId);
+
+      // Need to update microservice properties if we have modified or added properties of microservices.
+      Microservice microserviceTemp = new Microservice();
```
Comment (liubao68) [inline service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/task/MicroserviceRegisterTask.java:91]: When you update microservcie propeties, the client may not be notified for this change.  And why you implement this feature and does your scenario need notification?

File: service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/task/MicroserviceRegisterTask.java [hunk 6]
```diff
@@ -86,6 +86,24 @@ protected boolean doRegister() {
...
+      EnvAdapterManager.INSTANCE.processMicroserviceWithAdapters(microserviceTemp);
+      if (!microserviceTemp.getProperties().isEmpty()) {
+        Map<String, String> propertiesTemp = microserviceTemp.getProperties();
+        if (srClient.getMicroservice(serviceId) != null) {
+          microserviceTemp.setProperties(srClient.getMicroservice(serviceId).getProperties());
+          microserviceTemp.getProperties().putAll(propertiesTemp);
```
Comment (liubao68) [inline service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/task/MicroserviceRegisterTask.java:97]: srClient.getMicroservice(serviceId)  is a remote RPC call, use the result above

File: clients/service-center-client/src/test/java/org/apache/servicecomb/service/center/client/ServiceCenterClientTest.java (hunk 4 above)
Comment (Vientu) [reply clients/service-center-client/src/test/java/org/apache/servicecomb/service/center/client/ServiceCenterClientTest.java]: OK

File: service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/task/MicroserviceRegisterTask.java (hunk 5 above)
Comment (Vientu) [reply service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/task/MicroserviceRegisterTask.java:91]: To support the gray deployment function。

File: service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/task/MicroserviceRegisterTask.java (hunk 6 above)
Comment (Vientu) [reply service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/task/MicroserviceRegisterTask.java:98]: OK

**Review by liubao68 — APPROVED** [review]

**Pull request merged**